### PR TITLE
shift error run

### LINKED ISSUE #22: shift error in run
I received an error: Error in shift[i] <- score(bcorr, lags) : replacement has a length zero.

The command was: LPWC::corr.bestlag(kmat, timepoints = timepoints)

The first 5 rows of the matrix kmat[] are :
kmat[1:5,]
              k2.5m       k5m      k10m      k15m      k20m       k30m        k1h        k3h        k6h        k9h       k12h       k15h       k18h
AT1G50040 0.6762992 0.6357176 0.0000000 0.0000000 0.0000000 -1.0994969 -0.7865366  0.0000000 -0.7189853  0.0000000  0.0000000  0.0000000  0.0000000
AT3G54720 0.4488114 0.0000000 0.3747782 0.6030834 0.5920215  0.5217524  0.0000000 -0.6376395 -0.5126850 -0.5436959 -0.6493731  0.0000000 -0.2869364
AT1G01010 0.0000000 0.0000000 0.0000000 0.0000000 0.0000000  0.0000000  1.1150165  1.9223267  2.0520983  2.2537342  1.5281845  1.4821853  1.3771516
AT1G01030 0.0000000 0.0000000 0.7226280 0.6738165 0.4817093 -0.4557851  0.0000000  0.5577736  0.0000000 -0.7737706 -0.7789661 -0.4720915  0.0000000
AT1G01040 0.0000000 0.0000000 0.0000000 0.0000000 0.0000000  0.0000000 -0.6924797  0.0000000  0.2577919  0.6346532  0.2693469  0.1988710  0.0000000
 k21h       k24h       k27h      k30h
AT1G50040 -0.8190957 -0.7514720 -0.8236975 0.0000000
AT3G54720 -0.3000221 -0.3970144 -0.2185353 0.0000000
AT1G01010  1.5380587  1.5610146  0.6812520 0.6232566
AT1G01030  0.0000000  0.0000000  0.0000000 0.0000000
AT1G01040  0.0000000  0.0000000  0.0000000 0.0000000

 Timepoints are: 
timepoints
 [1]  0.000  0.042  0.083  0.170  0.250  0.330  0.500  1.000  3.000  6.000  9.000 12.000 15.000 18.000 21.000 24.000 27.000 30.000

  After entering the command, it ran for a while. I left if to go overnight and found the error in the morning.

Matthew

### PR DESCRIPTION
**Pull request by thevaachandereng**
closed #22

### CODE REVIEW

**Comment by thevaachandereng** [comment]
Please merge it when you are done checking. I am all done. We dont have to specify object names in hclust, it does it automatically. However, this wouldnt be fixed if you run the vignette unless you remove the code chunck with install_github, but please place it back as we need it for pkgdown to work.

**Comment by agitter** [comment]
Can you please review the code coverage report?  Do we need another test case to get back to 100% coverage?

**Comment by thevaachandereng** [comment]
Fixed it. Sorry, I missed that the codecov dropped.

File: tests/testthat/test_corr_bestlag.R [hunk 1]
```diff
@@ -12,8 +12,10 @@ test_that("the corr is", {
                timepoints = c(0, 2, 4, 6, 8, 18, 24, 32, 48, 72), penalty = "high")$lags[1]), 1)
   expect_equal(signif(corr.bestlag(data = simdata[20:30, ],
                timepoints = c(0, 2, 4, 6, 8, 18, 24, 32, 48, 72))$corr[5], 2), 0.91)
-  expect_error(signif(corr.bestlag(data = simdata,
-               timepoints = c(0, 2, 4, 6, 8, 18, 24, 32, 48, 72, 75))$corr[10], 2))
+  expect_equal(signif(corr.bestlag(data = simdata[20:30, ],
```
Comment (agitter) [inline tests/testthat/test_corr_bestlag.R]: The new lines 15-16 look the same as lines 13-14.  Did you mean to delete 13-14 and leave the test case that was deleted here?

File: vignettes/LPWC.Rmd [hunk 2]
```diff
@@ -80,9 +80,10 @@ library(LPWC)
 
 ## Required input
 
-**Data**:  Input to LPWC is a matrix.
+**Data**:  Input to LPWC is a matrix/data.frame.
 We assume here that it is a gene expression matrix, but other data types are suitable as well.
-The expression matrix should be n -by- p where n is the number of genes (in rows) and p is the number of timepoints.
+The expression matrix should be n -by- p where n is the number of genes (in rows) and p is the number of timepoints. The object names can be specified in the row names of ythe data frame 
+using \code{rownames}.
```
Comment (agitter) [inline vignettes/LPWC.Rmd]: Does the syntax `\code{}` work in vignettes as well?  If it follows Markdown syntax it won't, so I may change this.

File: vignettes/LPWC.Rmd [hunk 3]
```diff
@@ -80,9 +80,10 @@ library(LPWC)
...
 
 The object **simdata** is a simulated dataset for 200 genes with 8 timepoints that follow the biological [impulse model](https://doi.org/10.1093/bioinformatics/btw665).
 This is stored as a matrix.
```
Comment (agitter) [inline vignettes/LPWC.Rmd]: Should we be more precise here?  Isn't simdata actually a data frame with row and column names?

File: tests/testthat/test_corr_bestlag.R [hunk 4]
```diff
@@ -2,18 +2,20 @@ context("")
 test_that("the corr is", {
   expect_equal(signif(corr.bestlag(data = simdata[1:10, ],
                timepoints = c(0, 2, 4, 6, 8, 18, 24, 32, 48, 72))$corr[5], 2), 0.93)
-  expect_equal(corr.bestlag(data = simdata[50:70, ],
-               timepoints = c(0, 2, 4, 6, 8, 18, 24, 32, 48, 72), C = 10000000)$lags[1], 2)
-  expect_equal(corr.bestlag(data = simdata[50:70, ],
-               timepoints = c(0, 2, 4, 6, 8, 18, 24, 32, 48, 72), C = 10000000, penalty = "high")$lags[1], 2)
-  expect_equal(corr.bestlag(data = simdata[50:70, ],
-               timepoints = c(0, 2, 4, 6, 8, 18, 24, 32, 48, 72), penalty = "high")$lags[1], 1)
-  expect_equal(corr.bestlag(data = simdata[50:60, ],
```
[the remark this answers is not in the document]
Comment (thevaachandereng) [reply tests/testthat/test_corr_bestlag.R]: I think I accidentally removed this test case and kept the other extra one.

File: vignettes/LPWC.Rmd (hunk 3 above)
Comment (thevaachandereng) [reply vignettes/LPWC.Rmd]: We can be more precise. But if people load the package, they do understand it is a data.frame rather than matrix.

**Comment by thevaachandereng** [comment]
@agitter I have addressed the issues you commented on. Please check your orcid id. I just realize you can add your orcid id to an R package.

File: vignettes/LPWC.Rmd [hunk 5]
```diff
@@ -86,7 +86,7 @@ The expression matrix should be n -by- p where n is the number of genes (in rows
 If **data** is a data frame, the gene names can be specified using the `row.names()`.
 
 The object **simdata** is a simulated dataset for 200 genes with 8 timepoints that follow the biological [impulse model](https://doi.org/10.1093/bioinformatics/btw665).
-This is stored as a matrix.
+This is stored as a data frame in the package with row names representing gene labels.
```
Comment (agitter) [inline vignettes/LPWC.Rmd]: Thanks, this detailed phrasing is helpful.

File: tests/testthat/test_corr_bestlag.R [hunk 6]
```diff
@@ -2,18 +2,20 @@ context("")
...
+               timepoints = c(0, 2, 4, 6, 8, 18, 24, 32, 48, 72), C = 10000000, penalty = "high")$lags[1]), 2)
+  expect_equal(names(corr.bestlag(data = simdata[50:70, ], timepoints = c(0, 2, 4, 6, 8, 18, 24, 32, 48, 72),
+              C = 10000000, penalty = "high")$lags[1]), rownames(simdata[50, ]))
+  expect_equal(as.numeric(corr.bestlag(data = simdata[50:70, ],
+               timepoints = c(0, 2, 4, 6, 8, 18, 24, 32, 48, 72), penalty = "high")$lags[1]), 1)
+  expect_equal(as.numeric(corr.bestlag(data = simdata[50:60, ],
+               timepoints = c(0, 2, 4, 6, 8, 18, 24, 32, 48, 72), penalty = "low")$lags[1]), 1)
   expect_equal(signif(corr.bestlag(data = simdata[20:30, ],
                timepoints = c(0, 2, 4, 6, 8, 18, 24, 32, 48, 72))$corr[5], 2), 0.91)
-  expect_error(signif(corr.bestlag(data = simdata,
```
Comment (agitter) [inline tests/testthat/test_corr_bestlag.R]: Has this `expect_error` test case been removed?

**Comment by agitter** [comment]
> Please check your orcid id.

My ORCID is correct. Thanks for adding it.

I have one more test case question, but everything else looks good.  We could update the date in the DESCRIPTION as well so that once this is merged we have all of the files ready for CRAN submission.

**Comment by thevaachandereng** [comment]
Thanks @agitter. I removed the test case because we had a similar one on top. I forgot to comment on that. But, I guess it is okay to have multiple test cases.

**Pull request merged**